test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -914,6 +964,29 @@ public async Task WriteMetadataDocumentAsync_WorksForXmlCsdl()
                 "</edmx:Edmx>", payload);
         }
 
+        [Fact]
+        public async Task WriteLargeMetadataDocumentAsync_CallMultipleTimes_WorksForXmlCsdl()

```

Comment:
Did this test reproduce the exception from the original issue before your fix?

Reply:
@habbes Below is the test that will repro the issue if you do not apply this fix:

```cs
public async Task WriteLargeMetadataDocumentAsync_CalledMultipleTimes_WorksForXmlCsdl_NoExceptionThrown()
{
    // Arrange
    string contentType = "application/xml";
  
   ... // the rest of the code
```